src/lib.rs

```diff
@@ -327,11 +311,11 @@ impl ElasticClient for hyper::Client {
 
         match req.method {
             HttpMethod::Get => self.get(&url).headers(params.headers.to_owned()).send(),
-            HttpMethod::Post => req_with_body!(self, url, req, params, post),
+            HttpMethod::Post => self.post(&url).headers(params.headers.to_owned()).body(&**req.body.unwrap().into_owned()).send(),
```

Comment:
This is pure novice code - I wasn't aware I was copying - something I'd like to avoid. I haven't quite got my head around all this - odds are `Cursor` would work, but I've only now contextualized it. I'll go read up on `Cursor`